compiler/next/lib/frontend/frontend-queries.cpp

```diff
@@ -145,31 +145,40 @@ const LocationsMap& fileLocations(Context* context, UniqueString path) {
   // Get the result of parsing
   const uast::Builder::Result& p = parseFile(context, path);
   // Create a map of ast to Location
-  std::unordered_map<const ASTNode*, Location> result(p.locations.size());
+  LocationsMap result(p.locations.size());
   for (auto& pair : p.locations) {
-    if (pair.first != nullptr)
-      result.insert(pair); // Insert { ASTNode, Location } into result map
+    const ASTNode* ast = pair.first;
+    Location loc = pair.second;
+    if (ast != nullptr && !ast->id().isEmpty()) {
+      result.insert({ast->id(), loc});
+    }
   }
 
   return QUERY_END(result);
 }
 
-const Location& locate(Context* context, const ASTNode* ast) {
-  QUERY_BEGIN(locate, context, ast);
+const Location& locateID(Context* context, ID id) {
+  QUERY_BEGIN(locateID, context, id);
```

Comment:
Would it be possible to simplify `QUERY_BEGIN` if we used `__FUNCTION__` instead of explicitly writing out `locateID` (or the function name, in general)?

Reply:
I used to do that but we need to be able to re-run the query when a dependency needs it and as a result we have to be able to save the function pointer in the maps supporting the queries. `__FUNCTION__` `__func__` aren't sufficient for that part.